### PR TITLE
support annotations so oidc can be annotated

### PR DESCRIPTION
**Pull request by drewwells**

### CODE REVIEW

**Comment by faisal-memon** [comment]
Thanks @drewwells for submitting this fix.

File: charts/spire/charts/spiffe-oidc-discovery-provider/templates/deployment.yaml [hunk 1]
```diff
@@ -6,6 +6,10 @@ metadata:
   namespace: {{ include "spiffe-oidc-discovery-provider.namespace" . }}
   labels:
     {{- include "spiffe-oidc-discovery-provider.labels" . | nindent 4 }}
+  {{- if .Values.annotations }}
+  annotations:
+    {{- toYaml .Values.annotations | nindent 4 }}
+  {{- end }}
```
Comment (faisal-memon) [inline charts/spire/charts/spiffe-oidc-discovery-provider/templates/deployment.yaml:12]: The other templates follow a format like this one here from the [spire server statefulset](https://github.com/spiffe/helm-charts/blob/f461a017012a6268d8d1c996841111627b79497c/charts/spire/charts/spire-server/templates/statefulset.yaml#L31-L33). Would you mind changing this up so it matches?

Comment (faisal-memon) [reply charts/spire/charts/spiffe-oidc-discovery-provider/templates/deployment.yaml:12]:
```suggestion
  annotations:
    {{- with .Values.annotations }}
      {{- toYaml . | nindent 4 }}
    {{- end }}
```

Comment (kfox1111) [reply charts/spire/charts/spiffe-oidc-discovery-provider/templates/deployment.yaml:12]: @faisal-memon does line 11 there have too much indention now? seems like it has 6 but nindent is set to 4. might break on multiple annotations?

Comment (faisal-memon) [reply charts/spire/charts/spiffe-oidc-discovery-provider/templates/deployment.yaml:12]: indentation might be off

**Review by kfox1111 — APPROVED** [review]
LGTM

**Review by faisal-memon — APPROVED** [review]
Thanks @drewwells for making the requested changes.

**Pull request merged**